### ISSUE TITLE
Sign up page polish

### ISSUE THREAD

**Issue by jfly** [cubing/worldcubeassociation.org#421]
Some suggestions from @StachuDotNet:

- The visual separation of "I have competed in a comp" and "I haven't..." is bad right now
I kinda had to guess that that was a thing to click
![image](https://cloud.githubusercontent.com/assets/277474/13900975/ba7b7ae0-edd0-11e5-8e0e-56eae63dc895.png)
  - needs a border or background or something
- Also maybe explain /why/ I should sign up on the sign up page

**Comment by BiscuitCubed** [comment]
The other thing that could be done is asking "Have you competed in a WCA competition before?" Instead and just offering 'Yes.' and 'No.' I'll go see if I can figure out any changes that could help it.

**Comment by BiscuitCubed** [comment]
Here's a simple fix to the "Have you competed in a WCA competition" thing.

On the first one (on the li level) use this CSS.

background: #bdc3c7;
    width: 49.4%;
    display: inline-block;

And on the second one add this

    background: #bdc3c7;
    right: -5px;
    width: 49.4%;
    display: inline-block;

That should do what you want I think.

**Comment by jfly** [comment]
Thanks for looking into this! Could you add a screenshot of what you have in mind?

**Comment by BiscuitCubed** [comment]
I realized that I had the percentages off (it was a little different with the developer console open). I need to work on that for different screen sizes. They should be 49.6%

Here is a screenshot, and when you hover over the buttons it does change color. I also need to change some styles for when a link is clicked. 

![registration](https://cloud.githubusercontent.com/assets/16313471/13910878/38775514-eef7-11e5-8d9d-7cf928dd0b85.png)

**Comment by BiscuitCubed** [comment]
I think maybe adding a little bit in between the buttons and the message may work a little better.

**Comment by jfly** [comment]
Looking good, let's continue this discussion over a PR

**Comment by BiscuitCubed** [comment]
Sorry, I haven't really used github much. What's a PR?

**Comment by jfly** [comment]
Oh, PR stands for pull request! If you don't know how to make one, feel free to hit me up over gchat (you can find my Gmail address on my github profile, I'm pretty sure).

**Comment by BiscuitCubed** [comment]
Ah okay. I've only worked with git through the cmd line, and not much at that.

**Comment by BiscuitCubed** [comment]
I don't see your email...

**Comment by BiscuitCubed** [comment]
I'm assuming registrations.scss is the correct stylesheet to put stuff in?

**Comment by jfly** [comment]
I've updated https://github.com/jfly to show my email =)

**Comment by StachuDotNet** [comment]
Some feedback:
The color contrast in the above screenshot is a bit poor. Maybe a slightly
lighter background, or different colored text?

On Mon, Mar 21, 2016 at 2:19 AM, Jeremy Fleischman <notifications@github.com

**Comment by havijw** [comment]
I can take a shot at this! I think moving the "Please let us know..." text to before the response buttons makes sense, as well as making the buttons more distinct.

**Comment by FinnIckler** [comment]
Are you well versed in React? Would be the perfect opportunity to migrate the signup page to react.
It's in `WcaOnRails/app/views/devise/registrations/new.html.erb` right now. The only complicated parts I see is migrating simple form and switching the Country input to use the API endpoint instead of Ruby.
If you are not, feel free to ignore this and just fix the the content

**Comment by gregorbg** [comment]
Migrating this to React could be tricky because we use a library called [Devise](https://github.com/heartcombo/devise) for authentication (including creating new accounts). This library expects very precise POST requests, so whatever you do in React _has_ to mimic the existing solution in terms of using an HTML form. Otherwise Devise won't work.

**Comment by havijw** [comment]
I'm not super experienced with React so I'll leave the migration for another person/time.

I think switching to an accordion makes sense here:
![signup-collapsed](https://user-images.githubusercontent.com/44988777/219760292-be2db245-fb4f-49c1-a238-1907a39f7fa3.png)
![signup-have-competed](https://user-images.githubusercontent.com/44988777/219760289-b4d537b8-a9d1-435d-9407-e53a26886daf.png)
This makes the buttons more visible, and I like having the label above for consistency with the other form fields.

**Comment by dunkOnIT** [comment]
Looking great! One minor suggestion: End "...in a WCA competition before." with a colon ("before:") instead of a period, as another subtle prompt to the user to select an option.

**Comment by havijw** [comment]
Thanks! I agree about that change. Should I make that change in [config/locales/en.yml](https://github.com/thewca/worldcubeassociation.org/blob/39800aced7e1226c0f49d910a0e361fe25fd762d/WcaOnRails/config/locales/en.yml#L830)? If so, should I make corresponding changes for other locales?

**Comment by havijw** [comment]
Right now, I'm updating the javascript at the bottom of the page. I'm failing a few tests at the moment, including ["changing have competed and noobie, does not leak birthday information"](https://github.com/thewca/worldcubeassociation.org/blob/39800aced7e1226c0f49d910a0e361fe25fd762d/WcaOnRails/spec/features/sign_up_spec.rb#L235).

The existing version of the sign up page passes this test, but when I replicated the actions by hand, the birthday information actually is being leaked (in fact, every field in the "never competed" tab is filled out). I also did this on the actual (using my dad's WCA ID), and I could view his birthday by performing the actions in the test using his ID.

Is there something about this test that I'm not understanding? Should I create a new issue for this?

**Comment by notrichardpeng** [comment]
> Thanks! I agree about that change. Should I make that change in [config/locales/en.yml](https://github.com/thewca/worldcubeassociation.org/blob/39800aced7e1226c0f49d910a0e361fe25fd762d/WcaOnRails/config/locales/en.yml#L830)? If so, should I make corresponding changes for other locales?

Yes, please make changes to `en.yml` but no need to change other locales -- verified translators would update them on their own :).

**Comment by dunkOnIT** [comment]
@havijw have passed your other question re tests to the team, just waiting on feedback from them!

**Comment by havijw** [comment]
Thanks! I think I mostly solved the problem but there's still a small thing that I mentioned in [the PR](https://github.com/thewca/worldcubeassociation.org/pull/7649).
